src/integrationTest/java/com/mongodb/hibernate/JakartaPersistenceBootstrappingIntegrationTests.java

```diff
@@ -16,67 +16,48 @@
+import com.mongodb.client.MongoCollection;
+import com.mongodb.hibernate.junit.InjectMongoCollection;
+import com.mongodb.hibernate.junit.MongoExtension;
 import jakarta.persistence.Column;
 import jakarta.persistence.Entity;
-import jakarta.persistence.EntityManagerFactory;
 import jakarta.persistence.Id;
-import jakarta.persistence.Persistence;
 import jakarta.persistence.Table;
-import org.junit.jupiter.api.AutoClose;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.BeforeEach;
+import org.bson.BsonDocument;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.testing.orm.junit.EntityManagerFactoryScope;
+import org.hibernate.testing.orm.junit.Jpa;
+import org.hibernate.testing.orm.junit.Setting;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 
+@Jpa(
+        exportSchema = false,
+        integrationSettings =
+                @Setting(
+                        name = AvailableSettings.CONNECTION_PROVIDER,
+                        value = "com.mongodb.hibernate.jdbc.MongoConnectionProvider"),
+        annotatedClasses = JakartaPersistenceBootstrappingIntegrationTests.Item.class)
+@ExtendWith(MongoExtension.class)
 class JakartaPersistenceBootstrappingIntegrationTests {
 
-    private static MongoConfiguration config;
-
-    @AutoClose
-    private static EntityManagerFactory entityManagerFactory;
-
-    @AutoClose
-    private static MongoClient mongoClient;
-
-    @BeforeAll
-    static void beforeAll() {
-        entityManagerFactory = Persistence.createEntityManagerFactory("test-persistence-unit");
-        config = new MongoConfigurationBuilder(entityManagerFactory.getProperties()).build();
-        mongoClient = MongoClients.create(config.mongoClientSettings());
-    }
-
-    @BeforeEach
-    void beforeEach() {
-        mongoClient.getDatabase(config.databaseName()).drop();
-    }
+    @InjectMongoCollection("items")
+    private static MongoCollection<BsonDocument> collection;
 
     @Test
-    void smoke() {
-        try (var entityManager = entityManagerFactory.createEntityManager()) {
-            var transaction = entityManager.getTransaction();
-            try {
-                transaction.begin();
-                var item = new Item();
-                item.id = 1;
-                entityManager.persist(item);
-            } finally {
-                transaction.commit();
-            }
-            assertEquals(
-                    1,
-                    mongoClient
-                            .getDatabase(config.databaseName())
-                            .getCollection("items")
-                            .countDocuments());
-        }
+    void smoke(EntityManagerFactoryScope scope) {
+        scope.inTransaction(em -> {
+            var item = new Item();
+            item.id = 1;
+            em.persist(item);
+        });
+        assertThat(newArrayList(collection.find())).containsOnly(BsonDocument.parse("{_id: 1}"));
```

Comment:
`AssertJ` has util methods to convert `iterable` to `Collection`. I think its usage simplifies.

Reply:
See https://github.com/mongodb/mongo-hibernate/pull/63#discussion_r1988224528.

Reply:
good simplification! My only concern is maybe in the future different integration testing has different `MongoClient` settings, then more flexibility is desired. Perf issue is early opt currently to me. But for now I have no objection to your approach.

Reply:
My only change to your code is to use `https://junit.org/junit5/docs/current/user-guide` in lieu of `https://junit.org/junit5/docs/snapshot/user-guide`. Snapshot version is less desirable for it is unstable.

Reply:
I didn't touch other integration testings to avoid PR conflicts and labourious conflict resolving. We could simplify consistently in one single tech improvement PR in the future when all the PRs have been merged.

Reply:
> good simplification! My only concern is maybe in the future different integration testing has different `MongoClient` settings, then more flexibility is desired. Perf issue is early opt currently to me. But for now I have no objection to your approach.

nvm. I think this is a generic default approach. If some testing has special requirement, it simply avoid using this extension and go about programmatic configuration in ad-hoc way.

Reply:
> We could simplify consistently in one single tech improvement PR in the future when all the PRs have been merged.

OK, I'll create it later.